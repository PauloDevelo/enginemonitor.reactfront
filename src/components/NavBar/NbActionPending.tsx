import React, { useState, useEffect } from 'react';

import { FormattedMessage, defineMessages } from 'react-intl';

import actionManager from '../../services/ActionManager';

import jsonMessages from './NavBar.messages.json';

const navBarMsg = defineMessages(jsonMessages);


const NbActionPending = () => {
  const [nbAction, setActionCounter] = useState(0);

  useEffect(() => {
    actionManager.registerOnActionManagerChanged(setActionCounter);
    actionManager.countAction().then(setActionCounter);

    return () => {
      actionManager.unregisterOnActionManagerChanged(setActionCounter);
    };
  }, []);

  return <FormattedMessage {... navBarMsg.nbAction} values={{ nbAction }} />;
};

export default NbActionPending;
