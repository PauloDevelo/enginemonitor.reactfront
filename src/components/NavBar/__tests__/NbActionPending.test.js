import ignoredMessages from '../../../testHelpers/MockConsole';

import React from 'react';
import { mount } from 'enzyme';

import actionManager from '../../../services/ActionManager';
import NbActionPending from '../NbActionPending'

jest.mock('../../../services/ActionManager');

describe('Component NbActionPending', () =>{
    beforeAll(() => {
        ignoredMessages.length = 0;
        ignoredMessages.push("test was not wrapped in act(...)");
        ignoredMessages.push("[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry. Using default message as fallback.");
    });

    it('should render a specific message when there is no action to sync', () => {
        // Arrange
        const resp = 0;
        actionManager.countAction.mockResolvedValue(resp);

        // Act
        const wrapper = mount(<NbActionPending />);
        
        return Promise
            .resolve(wrapper)
            .then(() => {
                wrapper.update();

                // Assert
                expect(wrapper).toMatchSnapshot();

                expect(wrapper).toContainExactlyOneMatchingElement("FormattedMessage");
                expect(wrapper.text()).toBe("synced");
            });
    });

    it('should render a specific message when there is several action to sync', () => {
        // Arrange
        const resp = 3;
        actionManager.countAction.mockResolvedValue(resp);

        // Act
        const wrapper = mount(<NbActionPending />);

        return Promise
            .resolve(wrapper)
            .then(() => {
                wrapper.update();

                // Assert
                expect(wrapper).toMatchSnapshot();

                expect(wrapper).toContainExactlyOneMatchingElement("FormattedMessage");
                expect(wrapper.text()).toBe("3 actions to sync");
            });
    });

    it('should render a specific message when there is one action to sync', () => {
        // Arrange
        const resp = 1;
        actionManager.countAction.mockResolvedValue(resp);

        // Act
        const wrapper = mount(<NbActionPending />);

        return Promise
            .resolve(wrapper)
            .then(() => {
                wrapper.update();

                // Assert
                expect(wrapper).toMatchSnapshot();

                expect(wrapper).toContainExactlyOneMatchingElement("FormattedMessage");
                expect(wrapper.text()).toBe("1 action to sync");
            });
    });
});