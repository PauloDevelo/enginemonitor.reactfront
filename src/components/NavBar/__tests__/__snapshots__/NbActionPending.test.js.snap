// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component NbActionPending should render a specific message when there is no action to sync 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en-US"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
  timeZone="Asia/Kuala_Lumpur"
>
  <NbActionPending>
    <FormattedMessage
      defaultMessage="{nbAction, plural, =0 {synced} one {# action to sync} other {# actions to sync}}"
      description="Nb action pending"
      id="nbAction"
      values={
        Object {
          "nbAction": 0,
        }
      }
    >
      synced
    </FormattedMessage>
  </NbActionPending>
</IntlProvider>
`;

exports[`Component NbActionPending should render a specific message when there is one action to sync 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en-US"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
  timeZone="Asia/Kuala_Lumpur"
>
  <NbActionPending>
    <FormattedMessage
      defaultMessage="{nbAction, plural, =0 {synced} one {# action to sync} other {# actions to sync}}"
      description="Nb action pending"
      id="nbAction"
      values={
        Object {
          "nbAction": 1,
        }
      }
    >
      1 action to sync
    </FormattedMessage>
  </NbActionPending>
</IntlProvider>
`;

exports[`Component NbActionPending should render a specific message when there is several action to sync 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en-US"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
  timeZone="Asia/Kuala_Lumpur"
>
  <NbActionPending>
    <FormattedMessage
      defaultMessage="{nbAction, plural, =0 {synced} one {# action to sync} other {# actions to sync}}"
      description="Nb action pending"
      id="nbAction"
      values={
        Object {
          "nbAction": 3,
        }
      }
    >
      3 actions to sync
    </FormattedMessage>
  </NbActionPending>
</IntlProvider>
`;
